### PR TITLE
Documentation update to kitchen-terraform version 5.x

### PR DESCRIPTION
**Pull request by edwardbartholomew**
Update website Getting Started and AWS Provider (EC2) Tutorial with changes from version 5.

Update to source/layouts/layout.erb not included in #360 caused other tutorials to update

### CODE REVIEW

**Comment by edwardbartholomew** [comment]
Fixes #365

**Comment by edwardbartholomew** [comment]
Fixes #357

**Comment by edwardbartholomew** [comment]
@aaron-lane After rebasing, looks like one more of the tests have passed, one still failing though :(

**Comment by aaron-lane** [comment]
@edwardbartholomew no worries; that's to be expected. The deployment job will fail for any pull request originating from a fork. I've tried to just outright disable the job on non-master branches but haven't had success. I will try to review this and get it merged by the end of the week. :smile:

**Comment by edwardbartholomew** [comment]
Thank you Mr. Lane!

File: source/tutorials/amazon_provider_ec2.html.erb [hunk 1]
```diff
@@ -193,21 +193,24 @@ verifier:
...
           <% code("yml") do %>
-groups:
+systems:
   - name: default
     controls:
       - operating_system
-    hostnames: public_dns
-    username: ubuntu
           <% end %>
           Our group's name is default, and we expect to find a control file called <p class="font-weight-bold" style="color: #32c850; display: inline;">operating_system_spec.rb</p> within that group.
```
[the remark this answers is not in the document]
Comment (edwardbartholomew) [reply source/tutorials/amazon_provider_ec2.html.erb]: 👍

File: source/tutorials/amazon_provider_ec2.html.erb [hunk 2]
```diff
@@ -175,12 +174,13 @@ transport:
 verifier:
   name: terraform
   format: doc
```
[the remark this answers is not in the document]
Comment (edwardbartholomew) [reply source/tutorials/amazon_provider_ec2.html.erb]: Updated!  More information here for anyone following along at home: https://www.rubydoc.info/github/newcontext-oss/kitchen-terraform/Kitchen/Terraform/ConfigAttribute/Systems

File: source/tutorials/amazon_provider_ec2.html.erb [hunk 3]
```diff
@@ -68,14 +68,13 @@ touch main.tf variables.tf output.tf testing.tfvars
...
+          Add in the Kitchen-Terraform gem like below (substituting your Ruby version if necessary):
           <br><br>
           <% code("ruby") do %>
-ruby '2.3.1'
+ruby '2.6.5'
 
 source 'https://rubygems.org/' do
-  gem 'test-kitchen'
-  gem 'kitchen-terraform'
+  gem 'kitchen-terraform', '~> 5'
```
[the remark this answers is not in the document]
Comment (edwardbartholomew) [reply source/tutorials/amazon_provider_ec2.html.erb]: Great point!

**Comment by edwardbartholomew** [comment]
Thanks for the eyes @nictrix and @aaron-lane 

In terms of the extensive tutorial, we were missing the creation of below two files so I added steps in tutorial to create them:
* test/integration/extensive_suite/centos_attributes.yml 
* test/integration/extensive_suite/ubuntu_attributes.yml

**Comment by neofloww** [comment]
Can you put the tutorial link at the top of the README  before installation and/or by usage?
https://newcontext-oss.github.io/kitchen-terraform/tutorials/

**Comment by wwolfe** [comment]
In README 
Kitchen-Terraform provides a set of Test Kitchen plugins 
Test Kitchen link not working

**Comment by edwardbartholomew** [comment]
> Can you put the tutorial link at the top of the README before installation and/or by usage?
> https://newcontext-oss.github.io/kitchen-terraform/tutorials/

I think that's a great idea @neofloww 

> Can you put the tutorial link at the top of the README before installation and/or by usage?
> https://newcontext-oss.github.io/kitchen-terraform/tutorials/

I think that's a great idea, I'm always hunting for it myself! I'll take a look tomorrow along with that broken link you posted about.

**Comment by wwolfe** [comment]
in the readme for the example it says create file
./.kitchen.yml (Test Kitchen configuration)

when I run the example I get warning.
```
>>>>>> Message: Both /newcontext-oss/eds/kitchen.yml and /newcontext-oss/eds/.kitchen.yml found. Please use the un-dotted variant: /newcontext-oss/eds/kitchen.yml.
```
Might update to create kitchen.yml with out the "." prefix

**Comment by wwolfe** [comment]
Okay the example in the readme only works with terraform  0.11.14

  tfenv install 0.11.14

Also, I had to remove the host from main.tf

```
provider "docker" {
#  host    = "unix://localhost/var/run/docker.sock"
 }
```

**Comment by wwolfe** [comment]
on
https://github.com/newcontext-oss/kitchen-terraform/blob/master/CONTRIBUTING.md
bundle exec rake test:spec missing r
needs to be
bundle exec rake test:rspec

standard GitHub workflow.  link throws 404

**Comment by wwolfe** [comment]
for the actual tutorial in getting started
https://newcontext-oss.github.io/kitchen-terraform/getting_started.html
I would specify  terrform  ver 0.12.19   
both 11 and 12 work but the output is way shorter  with 12
Might even specify ruby-2.6.5

might add what success/ error looks like

```
       Success! The configuration is valid.

       Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
       Finished converging <kt-suite-terraform> (0m1.72s).
-----> Test Kitchen is finished. (0m3.00s)
```

```
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: 1 actions failed.
>>>>>>     Converge failed on instance <kt-suite-terraform>.  Please see .kitchen/logs/kt-suite-terraform.log for more details
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration
```

might make this change to kitchen.yml   
DEPRECATION: InSpec Attributes are being renamed to InSpec Inputs to avoid confusion with Chef Attributes. Use :inputs in your kitchen.yml verifier config instead of :attributes.
basic: Verifying

The error message above assumes you are now using kitchen.yml instead of .kitchen.yml
so you might change the getting started example to create kitchen.yml without the "." prefix

**Comment by wwolfe** [comment]
It looks like .kitchen.yml  could be updated in a lot of places to kitchen.yml

**Comment by wwolfe** [comment]
for ec2 tutorial I get an error .. may need some help
➜ bundle exec kitchen verify    # on the 1st verify command
-----> Starting Test Kitchen (v2.3.4)
-----> Verifying <default-ubuntu>...
$$$$$$ Running command `terraform workspace select kitchen-terraform-default-ubuntu` in directory /code/newcontext-oss/eds/tf_aws_cluster
$$$$$$ Running command `terraform output -json` in directory /code/newcontext-oss/eds/tf_aws_cluster
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: 1 actions failed.
>>>>>>     Verify failed on instance <default-ubuntu>.  Please see .kitchen/logs/default-ubuntu.log for more details
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration

at least destroy worked

**Comment by edwardbartholomew** [comment]
> In README
> Kitchen-Terraform provides a set of Test Kitchen plugins
> Test Kitchen link not working

> In README
> Kitchen-Terraform provides a set of Test Kitchen plugins
> Test Kitchen link not working

kitchen.ci domain expired this past week - Per Chef Community Slack, I see Chef is working with Fletcher Nichol to renew. I'm going to leave for now but we can revisit in future. Feel free to create issue if you'd like and we can track it.

**Comment by edwardbartholomew** [comment]
> in the readme for the example it says create file
> ./.kitchen.yml (Test Kitchen configuration)
> 
> when I run the example I get warning.
> 
> ```
> >>>>>> Message: Both /newcontext-oss/eds/kitchen.yml and /newcontext-oss/eds/.kitchen.yml found. Please use the un-dotted variant: /newcontext-oss/eds/kitchen.yml.
> ```
> 
> Might update to create kitchen.yml with out the "." prefix
> 
> update file tree hierarchy

kitchen-terraform will read either version of the file - I am guessing you might have both in current directory (maybe are you running from directory you have cloned this repo?)

**Comment by edwardbartholomew** [comment]
> Content regarding the transport plugin needs to be updated as it's no longer accurate: transport logic is delegated to InSpec. `verify.systems.*.key_files` will be necessary to use the AWS keypair.

I have made this change to the AWS EC2 Tutorial (Extensive tutorial was already updated). Already had my ssh key loaded in my agent when I was testing so didn't notice the change. So I think it's all ready for review again @aaron-lane . Will create issue for the README issues @wwolfe identified to be fixed in a different PR.

**Comment by neofloww** [comment]
for extensive_kitchen_terraform example mention it requires terraform 0.11 at the beginning.

**Comment by neofloww** [comment]
The kitchen.yml  file in the example
/tutorials/extensive_kitchen_terraform.html
for the Extensive Kitchen-Terraform example does not match the kitchen.yml in the example/extensive_kitchen_terraform directory and fails to parse as yml 
kitchen diagnose --no-instances --loader
errors out.
coping the one from the examples directory works
diff
```
➜ diff kitchen.yml ../extensive_kitchen_terraform/kitchen.yml
36,72d35
< #Populate the Test Kitchen configuration file, kitchen.yml , to match the following example.
<
< driver:
< #  name: terraform
<
<   # The test fixture Terraform configuration is configured to be the Terraform
<   # root module
<   root_module_directory: test/fixtures/wrapper
<
< provisioner:
<   name: terraform
<
< verifier:
<   name: terraform
<
< # Platforms provide hooks for overriding the global Test Kitchen plugin
< # configuration to provide platform-specific values
< platforms:
<   - name: centos
<
<     driver:
<       variables:
<         instances_ami: ami-ae7bfdb8
<
<         subnet_availability_zone: us-east-1a
<
<     verifier:
<       systems:
<         - name: local
<
<           # The customized_inspec_attribute InSpec attribute is configured to
<           # satisfy the inspec_attributes control
<           attrs_outputs:
<             customized_inspec_attribute: static_terraform_output
<
<           backend: local
<
101c64
<  - name: ubuntu
---
>   - name: ubuntu
111a75
>
149d112
< #  - # Kitchen-Terraform will assume that the InSpec profile for this suite is
151,152c114,116
<   - name: extensive_suite
<
---
>   - # Kitchen-Terraform will assume that the InSpec profile for this suite is
>     # located at test/integration/extensive_suite
>     name: extensive_suite

```

**Comment by wwolfe** [comment]
https://github.com/newcontext-oss/kitchen-terraform/pull/378#issuecomment-577012843
It looks like the issue above was a local configuration issue and is not an issue with the PR.

**Comment by wwolfe** [comment]
+1 Approve

**Comment by aaron-lane** [comment]
Let's move the fixes to the README to a separate branch. They're desperately needed and we can merge them presently.

**Comment by aaron-lane** [comment]
Thanks @edwardbartholomew @wwolfe @nictrix!

**Pull request merged**